### PR TITLE
Quick fix for #5754: Numbered list does not increment in ImageNode caption #5754

### PR DESCRIPTION
**Pull request by potatowagon**
**Before**:
see #5646 

**After**:

https://github.com/facebook/lexical/assets/19604232/d48f4880-364e-446b-91e1-44f47b0f2d3c

---

reason for failure: nested editor does not set the transforms of the nodes when creating the editor. This lead to updateChildrenListItemValue, which increments the value of `<li>` (ie, `1.`, `2`. ...), to not be called for nested editors.

- this is a short term fix
- did not add list plugin or markdown plugin to caption nested editor to keep size small
- the long term fix is something like https://github.com/facebook/lexical/tree/nestedsomething

### CODE REVIEW

File: packages/lexical-react/src/LexicalNestedComposer.tsx [hunk 1]
```diff
@@ -20,10 +20,20 @@ import {
...
 
+function getTransformSetFromKlass(
+  klass: KlassConstructor<typeof LexicalNode>,
+): Set<Transform<LexicalNode>> {
+  const transform = klass.transform();
+  return transform !== null
+    ? new Set<Transform<LexicalNode>>([transform])
+    : new Set<Transform<LexicalNode>>();
+}
+
```
Comment (potatowagon) [inline packages/lexical-react/src/LexicalNestedComposer.tsx:36]: this is from https://github.com/facebook/lexical/blob/5d61ab28edcd8ffe44e5411dcdf089e44dae4bc9/packages/lexical/src/LexicalEditor.ts#L493

File: packages/lexical-react/src/LexicalNestedComposer.tsx [hunk 2]
```diff
@@ -20,10 +20,20 @@ import {
   LexicalNode,
   LexicalNodeReplacement,
 } from 'lexical';
+import {KlassConstructor, Transform} from 'packages/lexical/src/LexicalEditor';
```
Comment (zurfyx) [inline packages/lexical-react/src/LexicalNestedComposer.tsx:23]: We can't use relative paths, these will not work well when we import these packages outside the playground. You want to use the name of the package instead

```suggestion
import type {KlassConstructor, Transform} from 'lexical';
```

**Review by zurfyx — APPROVED** [review]

**Comment by zurfyx** [comment]
Fix LGTM, thank you! Let's address the comment and feel free to merge

File: packages/lexical-react/src/LexicalNestedComposer.tsx (hunk 2 above)
Comment (potatowagon) [reply packages/lexical-react/src/LexicalNestedComposer.tsx]: thankyou good feedback :)

**Review by zurfyx — APPROVED** [review]

**Review by zurfyx — APPROVED** [review]

**Pull request merged**